src/index.js

```diff
@@ -1,17 +1,33 @@
 import { Engine } from "@babylonjs/core/Engines/engine";
-import createStartScene from "./startscene";
+import SpaceTruckerApplication from "./spaceTruckerApplication";
+import SpaceTruckerLoadingScreen from "./spaceTruckerLoadingScreen";
+import logger from "./logger";
+
 const CanvasName = "index-canvas";
+const launchButton = document.getElementById("btnLaunch");
+const pageLandingContent = document.getElementById("pageContainer");
 
-let canvas = document.createElement("canvas");
+const canvas = document.createElement("canvas");
 canvas.id = CanvasName;
-
 canvas.classList.add("background-canvas");
 document.body.appendChild(canvas);
 
-let eng = new Engine(canvas, true, null, true);
-let startScene = createStartScene(eng);
-eng.runRenderLoop(() => {
-    startScene.scene.render();
+const eng = new Engine(canvas, true, null, true);
+logger.logInfo("Created BJS engine");
+
+eng.loadingScreen = new SpaceTruckerLoadingScreen(eng);
+
+const theApp = new SpaceTruckerApplication(eng);    
+
+const btnClickEvtHandle = launchButton.addEventListener("click", () => {
```

Comment:
addEventListener does not return an object. If you want to remove it (which you should), you will need to create a function outside of this context and add/remove this function.

Reply:
och, nice catch.